libs/wingc/src/jsify.rs

```diff
@@ -1314,10 +1319,44 @@ impl<'a> JSifier<'a> {
 
 		let body = match &func_def.body {
 			FunctionBody::Statements(scope) => self.jsify_scope_body(scope, ctx),
-			FunctionBody::External(file_path) => {
+			FunctionBody::External(extern_path) => {
+				let extern_path = Utf8Path::new(extern_path);
+				let entrypoint_dir = if self.compilation_init_path.is_file() {
+					self.compilation_init_path.parent().unwrap()
+				} else {
+					self.compilation_init_path
+				};
+
+				// extern_path should always be a sub directory of entrypoint_dir
+				let rel_path = extern_path
+					.strip_prefix(&entrypoint_dir)
+					.expect(&format!("{extern_path} is not a sub directory of {entrypoint_dir}"));
```

Comment:
This would be a panic since it would be considered a bug in the compiler to have gotten to this point. That being said I will add a nicer check earlier

Reply:
Scratch that, this should be a diagnostic. I'll change it to not panic